<a href="https://colab.research.google.com/github/Rodrigo47589/-C8280/blob/main/Actividad1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Pregunta 1 : Mire en la parte superior e inferior de data.tsv entregado en esta actividad. Piensa en cómo se estructuran los datos. ¿Qué crees que significa cada columna? ¿Qué crees que significa cada fila?

In [12]:
!head data.tsv

order_id	quantity	item_name	choice_description	item_price
1	1	Chips and Fresh Tomato Salsa	NULL	$2.39 
1	1	Izze	[Clementine]	$3.39 
1	1	Nantucket Nectar	[Apple]	$3.39 
1	1	Chips and Tomatillo-Green Chili Salsa	NULL	$2.39 
2	2	Chicken Bowl	[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]	$16.98 
3	1	Chicken Bowl	[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]	$10.98 
3	1	Side of Chips	NULL	$1.69 
4	1	Steak Burrito	[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]	$11.75 
4	1	Steak Soft Tacos	[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]	$9.25 


1. Las columnas son las caracteristicas/atributos que describe un datos. Por ejemplo: order_id, quantity,  item_name, choice_description e item_price

2. Las filas significan el registro de datos en su respectiva columna. Por ejemplo:el numero de orden, el nombre del pedido,la cantidad del pedido, el precio del alimento y la descripción 

Pregunta 2: ¿Cuántos pedidos parece haber?

In [14]:
!tail data.tsv

1831	1	Chips	NULL	$2.15 
1831	1	Bottled Water	NULL	$1.50 
1832	1	Chicken Soft Tacos	[Fresh Tomato Salsa, [Rice, Cheese, Sour Cream]]	$8.75 
1832	1	Chips and Guacamole	NULL	$4.45 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Black Beans, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Guacamole, Lettuce]]	$11.25 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Lettuce]]	$8.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Lettuce]]	$8.75 



!tail: Muestra las ultimas lineas.

Con el uso de !tail, podemos ver que hay un total de 1834 pedidos 

Pregunta 3: ¿Cuántas líneas hay en este archivo?

In [8]:
!wc -l data.tsv

4624 data.tsv


!wc -l: Cuenta todas las lineas de un archivo 

Nos muestra que hay un total de 4624

Pregunta 4: ¿Qué burrito es más popular, steak o chicken?

In [9]:
!grep -i 'steak burrito' data.tsv | wc -l

368


In [10]:
!grep -i 'chicken burrito' data.tsv | wc -l

553


!grep -i: Especificar un patron del archivo contando las mayusculas y minusculas de ese patron como por ejemplo chicken burrito.

Y con el wc -l contamos la lineas de ese patron.

Vemos que el chicken burritos tiene mas lineas que el steak burrito.Es decir, el chicken burrito es el mas popular. 

Pregunta 5: ¿Los burritos de pollo suelen tener frijoles negros o frijoles pintos?

In [15]:
!grep -i 'chicken burrito' data.tsv | grep -i 'black beans' | wc -l

282


In [16]:
!grep -i 'chicken burrito' data.tsv | grep -i 'pinto beans' | wc -l

105


Para especificar el chicken burrito con los frijoles negros y frijoles pintos. Del anterior codigo, le agregamos otro !grep -i para que el patron especificado, en este caso el chicken burrito, podamos buscar su caracteristica especificada. Por ejemplo: Chicken burrito with black beans (Burrito de pollo con frijoles negros)

Y vemos que el black beans tiene mas lineas que el pinto beans. Es decir, los chicken burritos suelen tener mas black beans que pinto beans.

